pkg/apps/blackduck/latest/creater.go

```diff
@@ -153,13 +91,31 @@ func (hc *Creater) Ensure(blackduck *v1.Blackduck) error {
 		}
 	}
 
-	// Ensure
-	commonConfig := crdupdater.NewCRUDComponents(hc.KubeConfig, hc.KubeClient, hc.Config.DryRun, blackduck.Spec.Namespace, cpList, "app=blackduck,component!=postgres")
+	// Get non postgres components
+	cpList, err := hc.getComponents(blackduck)
+	if err != nil {
+		return err
+	}
+
+	// deploy non postgres and uploadcache component
+	commonConfig := crdupdater.NewCRUDComponents(hc.KubeConfig, hc.KubeClient, hc.Config.DryRun, blackduck.Spec.Namespace,
+		cpList, "app=blackduck,component notin (postgres,uploadcache)")
 	errors := commonConfig.CRUDComponents()
 	if len(errors) > 0 {
-		return fmt.Errorf("unable to update components due to %+v", errors)
+		return fmt.Errorf("unable to update non postgres and uploadcache components due to %+v", errors)
 	}
 
+	// log.Debugf("created/updated non postgres and upload cache component for %s", blackduck.Spec.Namespace)
+
+	// deploy upload cache component
+	commonConfig = crdupdater.NewCRUDComponents(hc.KubeConfig, hc.KubeClient, hc.Config.DryRun, blackduck.Spec.Namespace,
```

Comment:
Why aren't we deploying uploadcache and everything else (except postgres) at the same time?

Reply:
this is bcoz, if there is any change in seal key, it need to restart only the upload cache container